tuf/client/updater.py

```diff
@@ -3241,6 +3247,41 @@ def download_target(self, target, destination_directory,
         to ensure snapshot consistency).
         Default is True.
 
+     custom_download_handler:
+        A user provided function performing the actual target file download.
+        In order to comply with the TUF specification, the function implementation
+        should match the following description:
+
+        def download_handler_func(url, required_length)
+          <Purpose>
+            Given the 'url' and 'required_length' of the desired file, open a connection
+            to 'url', download it, and return the contents of the file.  Also ensure
+            the length of the downloaded file matches 'required_length' exactly.
+
+          <Arguments>
+            url:
+              A URL string that represents the location of the file.
+
+            required_length:
+              An integer value representing the length of the file.  This is an exact
+              limit.
+
+          <Side Effects>
+            A temprorary file object is created to store the contents of 'url'.
+
+          <Exceptions>
+            DownloadLengthMismatchError, if there was a
+            mismatch of observed vs expected lengths while downloading the file.
```

Comment:
Yes, this is the "_deliberately redundant implementation designed to complement tuf.download.\_check\_downloaded\_length()_"
https://github.com/theupdateframework/tuf/pull/1171#discussion_r510943049

Seems like it can nicely complement `custom_download_handler` too without the redundancy part!